evm/contract.go

```diff
@@ -0,0 +1,418 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package evm
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"reflect"
+	"strings"
+
+	"github.com/ava-labs/avalanche-tooling-sdk-go/utils"
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var ErrFailedReceiptStatus = fmt.Errorf("failed receipt status")
+
+func removeSurroundingParenthesis(s string) (string, error) {
+	s = strings.TrimSpace(s)
+	if len(s) > 0 {
```

Comment:
nit: assign len(s) to variable so it's calculated once( better perf)  and makes code easier to read

Reply:
not so sure it is easier to read.